server/api/teams.go

```diff
@@ -257,3 +259,101 @@ func (a *API) handleGetTeamUsers(w http.ResponseWriter, r *http.Request) {
 	auditRec.AddMeta("userCount", len(users))
 	auditRec.Success()
 }
+
+func (a *API) handleGetTeamUsersByID(w http.ResponseWriter, r *http.Request) {
+	// swagger:operation POST /teams/{teamID}/users getTeamUsersByID
+	//
+	// Returns a user[]
+	//
+	// ---
+	// produces:
+	// - application/json
+	// parameters:
+	// - name: teamID
+	//   in: path
+	//   description: Team ID
+	//   required: true
+	//   type: string
+	// security:
+	// - BearerAuth: []
+	// responses:
+	//   '200':
+	//     description: success
+	//     schema:
+	//       type: array
+	//       items:
+	//         "$ref": "#/definitions/User"
+	//   default:
+	//     description: internal error
+	//     schema:
+	//       "$ref": "#/definitions/ErrorResponse"
+
+	requestBody, err := io.ReadAll(r.Body)
+	if err != nil {
+		a.errorResponse(w, r, err)
+		return
+	}
+
+	var userIDs []string
+	if err = json.Unmarshal(requestBody, &userIDs); err != nil {
+		a.errorResponse(w, r, err)
+		return
+	}
+
+	auditRec := a.makeAuditRecord(r, "getUsersListByID", audit.Fail)
+	defer a.audit.LogRecord(audit.LevelAuth, auditRec)
+
+	vars := mux.Vars(r)
+	teamID := vars["teamID"]
+	userID := getUserID(r)
+
+	if !a.permissions.HasPermissionToTeam(userID, teamID, model.PermissionViewTeam) {
+		a.errorResponse(w, r, model.NewErrPermission("access denied to team"))
+		return
+	}
+
+	var users []*model.User
+	var error error
+
+	if len(userIDs) == 0 {
+		a.errorResponse(w, r, model.NewErrBadRequest("User IDs are empty"))
+		return
+	}
+
+	if userIDs[0] == model.SingleUser {
+		ws, _ := a.app.GetRootTeam()
+		now := utils.GetMillis()
+		user := &model.User{
+			ID:       model.SingleUser,
+			Username: model.SingleUser,
+			Email:    model.SingleUser,
+			CreateAt: ws.UpdateAt,
+			UpdateAt: now,
+		}
+		users = append(users, user)
+	} else {
+		users, error = a.app.GetUsersList(userIDs)
```

Comment:
Should we have pagination here?  If so we could roll that into the existing ticket on adding pagination to all APIs that need it.

Reply:
Yes, we should.